From c40fc9c4f4e2f850bbdf74f66c9444241a3b84e2 Mon Sep 17 00:00:00 2001
From: Raphael Mounier <mounierr07@gmail.com>
Date: Sat, 17 Jun 2023 16:48:11 +0200
Subject: [PATCH] add mono sim option

Disable the second sim by setting persist.sys.phh.monosim property
---
 phh-prop-handler.sh | 30 ++++++++++++++++++++++++++++++
 vndk.rc             |  3 +++
 2 files changed, 33 insertions(+)

diff --git a/phh-prop-handler.sh b/phh-prop-handler.sh
index 95cc07b..64d2a51 100644
--- a/phh-prop-handler.sh
+++ b/phh-prop-handler.sh
@@ -318,3 +318,33 @@ if [ "$1" == "sys.phh.dump_logs" ]; then
     exit
 fi
 
+#monosim
+if [ "$1" == "persist.sys.phh.monosim" ]; then
+    if [[ "$prop_value" != "false" && "$prop_value" != "true" ]]; then
+        exit 1
+    fi
+
+    if [ "$prop_value" == "true" ]; then
+		resetprop_phh persist.radio.multisim.config single
+		resetprop_phh persist.dsds.enabled false
+		resetprop_phh persist.radio.modem.cap 09995
+
+		resetprop_phh ro.config.modem_number 1
+		resetprop_phh ro.config.client_number 1
+
+		resetprop_phh ro.multi.rild false
+		resetprop_phh persist.odm.dsds.enabled false
+		resetprop_phh ro.odm.multi.rild false
+
+	else
+		resetprop_phh persist.radio.multisim.config dsds
+		resetprop_phh persist.dsds.enabled true
+		resetprop_phh persist.radio.modem.cap 09A9D
+
+		resetprop_phh ro.multi.rild true
+		resetprop_phh persist.odm.dsds.enabled true
+		resetprop_phh ro.odm.multi.rild true
+    fi
+    exit
+fi
+
diff --git a/vndk.rc b/vndk.rc
index 838cfe0..7225736 100644
--- a/vndk.rc
+++ b/vndk.rc
@@ -61,6 +61,9 @@ on property:persist.sys.phh.disable_a2dp_offload=1
 on property:persist.sys.phh.disable_a2dp_offload=*
     setprop persist.bluetooth.bluetooth_audio_hal.disabled ${persist.sys.phh.disable_a2dp_offload}
 
+on property:persist.sys.phh.monosim=*
+    exec u:r:phhsu_daemon:s0 root -- /system/bin/phh-prop-handler.sh "persist.sys.phh.monosim"
+
 on property:persist.sys.phh.root=*
     exec u:r:phhsu_daemon:s0 root -- /system/bin/phh-prop-handler.sh "persist.sys.phh.root"
 
-- 
2.25.1

